packages/xrpl/test/integration/transactions/batch.test.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  generateFundedWallet,
+  testTransaction,
+  verifySubmittedTransaction,
+} from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('Batch', function () {
+  let testContext: XrplIntegrationTestContext
+  let destination: Wallet
+  let wallet2: Wallet
+
+  async function testBatchTransaction(
+    batch: Batch,
+    wallet: Wallet,
+    retry?: {
+      count: number
+      delayMs: number
+    },
+  ): Promise<void> {
+    await testTransaction(testContext.client, batch, wallet, retry)
+    const promises: Array<Promise<void>> = []
+    for (const rawTx of batch.RawTransactions) {
+      promises.push(
+        verifySubmittedTransaction(testContext.client, rawTx.RawTransaction),
+      )
+    }
+    await Promise.all(promises)
+  }
+
+  beforeAll(async () => {
+    testContext = await setupClient(serverUrl)
+    wallet2 = await generateFundedWallet(testContext.client)
+    destination = await generateFundedWallet(testContext.client)
+  }, TIMEOUT)
+  afterAll(async () => teardownClient(testContext))
+
+  it(
+    'base',
+    async () => {
+      const tx: Batch = {
+        TransactionType: 'Batch',
+        Account: testContext.wallet.classicAddress,
+        Flags: BatchFlags.tfAllOrNothing,
+        RawTransactions: [
+          {
+            RawTransaction: {
+              TransactionType: 'Payment',
+              Flags: 0x40000000,
+              Account: testContext.wallet.classicAddress,
+              Destination: destination.classicAddress,
+              Amount: '10000000',
+            },
+          },
+          {
+            RawTransaction: {
+              TransactionType: 'Payment',
+              Flags: 0x40000000,
+              Account: testContext.wallet.classicAddress,
+              Destination: destination.classicAddress,
+              Amount: '10000000',
+            },
+          },
+        ],
+      }
+      const autofilled = await testContext.client.autofill(tx)
+      await testBatchTransaction(autofilled, testContext.wallet)
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'batch multisign',
+    async () => {
+      const tx: Batch = {
+        TransactionType: 'Batch',
+        Account: testContext.wallet.classicAddress,
+        Flags: BatchFlags.tfAllOrNothing,
+        RawTransactions: [
+          {
+            RawTransaction: {
+              TransactionType: 'Payment',
+              Flags: 0x40000000,
+              Account: testContext.wallet.classicAddress,
+              Destination: destination.classicAddress,
+              Amount: '10000000',
+            },
+          },
+          {
+            RawTransaction: {
+              TransactionType: 'Payment',
+              Flags: 0x40000000,
+              Account: wallet2.classicAddress,
+              Destination: destination.classicAddress,
+              Amount: '10000000',
+            },
+          },
+        ],
+      }
+      const autofilled = await testContext.client.autofill(tx, 2)
```

Comment:
Since `testContext.wallet` is the sender of the outer-batch-transaction, only `wallet2` needs to be included in the `BatchSigners` array.
```suggestion
      const autofilled = await testContext.client.autofill(tx, 1)
```

Reply:
fixed